src/test/scala/cookbook/RegOfVec.scala

```diff
@@ -0,0 +1,26 @@
+// See LICENSE for license details.
+
+package cookbook
+
+import chisel3._
+
+/* ### How do I create a Reg of type Vec?
+ *
+ * For information, please see the API documentation
+ * (https://chisel.eecs.berkeley.edu/api/index.html#chisel3.core.Vec)
+ */
+class RegOfVec extends CookbookTester(2) {
+  // Reg of Vec of 32-bit UInts
+  val regOfVec = Reg(Vec(4, UInt(32.W)))
```

Comment:
I can add it here, it just requires a little more explanation since it requires using a Vec.apply method that constructs a Wire (which is kind of confusing in the *type* vs *value* distinction that I'm trying to make more explicit). It's probably wise to have that example though since it is a cookbook after all.